src/FileFormats/LP/LP.jl

```diff
@@ -285,9 +285,472 @@ end
+const _COMMENT_REG = r"(.*?)\\(.*)"
+function _strip_comment(line::String)
+    if occursin("\\", line)
+        m = match(_COMMENT_REG, line)
+        return strip(String(m[1]))
+    else
+        return strip(line)
+    end
+end
+
+# a list of section keywords in lower-case
+const _KEYWORDS = Dict(
+    "max" => Val{:obj},
+    "maximize" => Val{:obj},
+    "maximise" => Val{:obj},
+    "maximum" => Val{:obj},
+    "min" => Val{:obj},
+    "minimize" => Val{:obj},
+    "minimise" => Val{:obj},
+    "minimum" => Val{:obj},
+    "subject to" => Val{:constraints},
+    "such that" => Val{:constraints},
+    "st" => Val{:constraints},
+    "s.t." => Val{:constraints},
+    "bounds" => Val{:bounds},
+    "bound" => Val{:bounds},
+    "gen" => Val{:integer},
+    "general" => Val{:integer},
+    "generals" => Val{:integer},
+    "bin" => Val{:binary},
+    "binary" => Val{:binary},
+    "binaries" => Val{:binary},
+    "end" => Val{:quit},
+)
+
+const _SENSE_ALIAS = Dict(
+    "max" => MOI.MAX_SENSE,
+    "maximize" => MOI.MAX_SENSE,
+    "maximise" => MOI.MAX_SENSE,
+    "maximum" => MOI.MAX_SENSE,
+    "min" => MOI.MIN_SENSE,
+    "minimize" => MOI.MIN_SENSE,
+    "minimise" => MOI.MIN_SENSE,
+    "minimum" => MOI.MIN_SENSE,
+)
+
+const _SUBJECT_TO_ALIAS = ["subject to", "such that", "st", "s.t."]
+
+const _CONSTRAINT_SENSE = Dict(
+    "<" => :le,
+    "<=" => :le,
+    "=" => :eq,
+    "==" => :eq,
+    ">" => :ge,
+    ">=" => :ge,
+)
+
+function _verify_name(variable::String, maximum_length::Int)
+    if length(variable) > maximum_length
+        return false
+    end
+    # m = match(START_REG, variable)
+    # if !isnothing(m)
+    #     return false
+    # end
+    m = match(NAME_REG, variable)
+    if m !== nothing
+        return false
+    end
+    return true
+end
+
+mutable struct CacheLPModel
+    objective_function::MOI.ScalarAffineFunction
+    linear_constraint_function::MOI.ScalarAffineFunction
+    linear_constraint_set::MOI.AbstractScalarSet
+    linear_constraint_open::Bool
+    linear_constraint_name::String
+    num_linear_constraints::Int
+    function CacheLPModel()
+        return new(
+            MOI.ScalarAffineFunction(MOI.ScalarAffineTerm{Float64}[], 0.0),
+            MOI.ScalarAffineFunction(MOI.ScalarAffineTerm{Float64}[], 0.0),
+            MOI.EqualTo(0.0),
+            false,
+            "",
+            0,
+        )
+    end
+end
+
+_set_sense!(T, model::Model, line) = nothing
+function _set_sense!(::Type{Val{:obj}}, model::Model, line)
+    return MOI.set(model, MOI.ObjectiveSense(), _SENSE_ALIAS[lowercase(line)])
+end
+
+function _add_new_variable!(model::Model, name::String)
+    var = MOI.add_variable(model)
+    MOI.set(model, MOI.VariableName(), var, name)
+    return var
+end
+
+function _get_variable_from_name(model::Model, variable_name::String)
+    variables_inside_model = MOI.get(model, MOI.ListOfVariableIndices())
+    for var_inside_model in variables_inside_model
```

Comment:
It might be worth constructing some benchmarks so you can profile this reader. This function is going to be very expensive. We should probably cache a dictionary mapping names to variable indices.